src/js/components/List.jsx

```diff
@@ -0,0 +1,197 @@
+import React, { PropTypes } from 'react';
+import KeyCodes from '../KeyCodes';
+
+type Props = {
+    className?: string,
+    children?: ReactChildren,
+    style?: Object,
+    data?: Array<Object>,
+    labelFunction?: Function,
+    defaultStyles: bool,
+    defaultContainerClass: bool,
+    defaultItemClass: bool,
+    defaultSelection?: Object,
+    onItemSelect?: Function,
+}
+
+type State = {
+    selectedItem?: Object,
+};
+
+// TODO: fix a few flow errors
+/*
+type RendererProps = {
+    listIndex: number,
+    listItem: string | Object,
+    labelFunction: Function,
+}
+*/
+
+
+export class List extends React.Component {
+
+    props: Props;
+    state: State;
+
+    static defaultProps: Props = {
+        style: {},
+        defaultStyles: true,
+        defaultContainerClass: true,
+        defaultItemClass: true,
+    };
+
+    constructor(props:Props) {
+        super(props);
+
+        this.state = {
+            selectedItem: null,
+        };
+    }
+
+    componentWillReceiveProps(nextProps:Props) {
+        if (this.props.data !== nextProps.data) {
+            this.setState({
+                selectedItem: null,
+            });
+        }
+
+        this._defaultSelection(nextProps);
+    }
+
+    componentDidMount() {
+        this._defaultSelection(this.props);
+    }
+
+    get selectedIndex():number {
+        return this.props.data ? this.props.data.indexOf(this.state.selectedItem) : -1;
+    }
+
+    get selectedItem():Object {
+        return this.state.selectedItem;
+    }
+
+    _defaultSelection(nextProps:Props) {
+        if (!this.state.selectedItem && nextProps.defaultSelection) {
+            this.setState({
+                selectedItem: nextProps.defaultSelection,
+            });
+        }
+    }
+
+    _onClickListItem(event:Event, index:number, item:Object) {
+        event.preventDefault();
+
+        this._selectItem(index, item);
+    }
+
+    _onKeyPressListItem(event:React$Event, index:number, item:Object) {
+        event.preventDefault();
+
+        if (event.nativeEvent.keyCode === KeyCodes.SPACEBAR) {
```

Comment:
re: keyboard I agree, having to tab through every item is pretty poor. I might actually use "hidden" radio buttons to get the right accessibility, since OS X has some significant default behavior for how keyboard works and I want to respect that without significant custom code.

re: selection, it seems a reasonable default behavior to me. If we discard it, are you suggesting that "selectedItem" becomes a prop that the developer needs to populate? I'm a little torn on that since the overwhelming use case for Lists is to allow something to be selected.